packages/wrangler/src/r2/helpers.ts

```diff
@@ -31,11 +31,12 @@ export async function listR2Buckets(
  */
 export async function createR2Bucket(
 	accountId: string,
-	bucketName: string
+	bucketName: string,
+	regionHint?: string
 ): Promise<void> {
 	return await fetchResult<void>(`/accounts/${accountId}/r2/buckets`, {
 		method: "POST",
-		body: JSON.stringify({ name: bucketName }),
+		body: JSON.stringify({ name: bucketName, regionHint }),
```

Comment:
The actual implementation isn't released yet, but I'll make sure to hold this PR till it is.